### PR TITLE
add type hint in pipeline model argument

### PR DESCRIPTION
**Pull request by y3sar**
# What does this PR do?

This PR adds type hints for model argument in pipeline function

## Who can review?
@Narsil 
@amyeroberts

### CODE REVIEW

File: src/transformers/pipelines/__init__.py [hunk 1]
```diff
@@ -505,7 +505,7 @@ def clean_custom_task(task_info):
 
 def pipeline(
     task: str = None,
-    model: Optional = None,
+    model: Optional[Union[str, Path]] = None,
```
Comment (amyeroberts) [inline src/transformers/pipelines/__init__.py:508]: I think the instances are `str`, `PreTrainedModel` or `TFPreTrainedModel`

**Review by amyeroberts — COMMENTED** [review]
@y3sar - thanks for adding this improvement! 

I think the hint should show that `model` can be a model instance too

**Review by amyeroberts — COMMENTED** [review]
@y3sar - thanks for adding this improvement! 

I think the hint should show that `model` can be a model instance too

File: src/transformers/pipelines/__init__.py (hunk 1 above)
Comment (y3sar) [reply src/transformers/pipelines/__init__.py:508]: thanks for the fast reply. So Path object shouldn't be there?

**Comment by y3sar** [comment]
I think I should add PretrainedModel and TFPretrainedModel in string form just like PreTrainedTokenizerFast was given in tokenizer argument. TYPE_CHECKING is false by default

**Comment by Narsil** [comment]
LGTM.

**Review by amyeroberts — APPROVED** [review]
Thanks for adding and iterating!

**Pull request merged**